services/blockchain-indexer/shared/dataService/business/pos/stakers.js

```diff
@@ -74,17 +76,40 @@ const getStakers = async params => {
 
 		const { address } = await getIndexedAccountInfo({ name, limit: 1 }, ['address']);
 		params.validatorAddress = address;
-		stakers.meta.validator.name = name;
+		stakersResponse.meta.validator.name = name;
 	}
 
 	// If validatorAddress is unavailable, return empty response
-	if (!params.validatorAddress) return stakers;
-
-	// TODO: Use count method directly once support for custom column-based count added https://github.com/LiskHQ/lisk-service/issues/1188
-	const [{ numStakers }] = await stakesTable.rawQuery(`SELECT COUNT(stakerAddress) as numStakers from ${stakesIndexSchema.tableName} WHERE validatorAddress='${params.validatorAddress}'`);
+	if (!params.validatorAddress) return stakersResponse;
+
+	// Search param is used to filter stakers by name. Prepare list of valid staker addresses
+	// for the given search (staker name) param
+	const allowedStakersAddressNameMap = {};
+	const stakerAddressQueryFilter = {};
+	if (params.search) {
+		const stakerAccountsInfo = await accountsTable.find(
+			{
+				search: {
+					property: 'name',
+					pattern: params.search,
+				},
+			},
+			['name', 'address'],
+		);
+
+		stakerAccountsInfo.forEach(accountInfo => {
+			allowedStakersAddressNameMap[accountInfo.address] = accountInfo.name;
+		});
+		stakerAddressQueryFilter.whereIn = {
+			property: 'stakerAddress',
+			values: Object.keys(allowedStakersAddressNameMap),
+		};
+	}
```

Comment:
Let us discuss offline the optimization that can be implemented here.

Reply:
Discussed offline.